docs/content/manual/manual.md

```diff
@@ -0,0 +1,5424 @@
+:::
+
+Note too that `.` is the input value at the particular stage
+in a "pipeline", specifically: where the `.` expression appears.
+Thus `.a | . | .b` is the same as `.a.b`, as the `.` in the
+middle refers to whatever value `.a` produced.
+
+::: Examples
+
+~~~
+.[] | .name
+[{"name":"JSON", "good":true}, {"name":"XML", "good":false}]
+"JSON"
+"XML"
+~~~
+
+:::
+
+## Function call
+
+jq provides [many builtin functions](#builtin-functions) for a variety of tasks,
+and you can also [define your own functions](#definitions).
+
+Each function has an _arity_ that specifies how many _arguments_ that function takes.
+For example,
+the function [`length`](#length) does not take any argument, so its arity is 0, whereas
+the function [`contains`](#contains) takes one argument, so its arity is 1.
+There may be several functions that have the same name, but different arities;
+for example, there exist two functions called `add`, taking zero and one arguments, respectively.
+To unambiguously identify a function `f` with arity `n`, we write `f/n`,
+e.g. `length/0`, `contains/1`, `add/0`, and `add/1`.
+
+To call a function `f` with no arguments (arity 0), we write `f`.
+To call a function `f` with `n` arguments (arity greater than zero), we write `f(a1; ...; an)`.
+
+::: Note
+Function calls use semicolons `;` instead of commas `,` to separate arguments,
+because `,` is already used for [concatenation](#concatenation).
+:::
+
+::: Examples
+
+The first example calls the function [`length/0`](#length):
+
+~~~
+length
+[1, 1, 2, 3]
+4
+~~~
+
+The next example calls
+[`range/2`](#range) and
+[`add/1`](#add):
+
+~~~
+add(range(0; .))
+2
+3
+~~~
+
+The last example calls [`while/2`](#while):
+
+~~~
+while(length < 3; . + "a")
+""
+""
+"a"
+"aa"
+~~~
+
+:::
+
+## Parenthesis
+
+Parentheses act as a grouping operator just as in any typical programming language.
+
+::: Examples
+
+~~~
+(. + 2) * 5
+1
+15
+~~~
+
+:::
+
+## Recursive descent: `..` {#recursive-descent}
+
+Recursively descends `.`, producing every value.  This is the
+same as [the zero-argument `recurse` function](#recurse).
+This is intended to resemble the XPath `//` operator.
+Note that `..a` does not work; use `.. | .a` instead.
+In the example below we use `.. | .a?` to find
+all the values of object keys "a" in any object found "below" `.`.
+
+This is particularly useful in conjunction with
+[`path(EXP)`](#path) and the
+[`?` operator](#error-suppression).
+
+::: Examples
+
+~~~
+.. | .a?
+[[{"a":1}]]
+1
+~~~
+
+:::
+
+
+# Paths
+
+In this section, we will show three very frequently used operators, namely for
+[iteration](#iteration-operator),
+[indexing](#indexing-operator), and
+[slicing](#slicing-operator).
+These operators serve to obtain parts of values.
+Furthermore, we will see how to [combine these operators](#compound-paths).
+
+## Iteration operator: `.[]` {#iteration-operator}
+
+The operator `.[]` returns the values contained inside the input value.
```

Comment:
I think it's best to consider "returns" and "outputs" as synonyms; this allows writing:

> Given two filters `f` and `g`, their concatenation
`f, g` first returns the outputs of `f`, then the outputs of `g`.

If we say that "returns" means "outputs at most one value", then this would need to be written as:

> Given two filters `f` and `g`, their concatenation
`f, g` first outputs the outputs of `f`, then the outputs of `g`.

And that feels quite awkward to me.

Reply:
Yes agree